src/components/molecules/CameraMicrophoneControls/CameraMicrophoneControls.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback } from "react";
+import classNames from "classnames";
+import Video, {
+  AudioTrackPublication,
+  VideoTrackPublication,
+} from "twilio-video";
+
+import { ContainerClassName } from "types/utility";
+
+import { VideoOverlayButton } from "components/atoms/VideoOverlayButton";
+
+import "components/molecules/CameraMicrophoneControls/CameraMicrophoneControls.scss";
+
+export interface CameraMicrophoneControlsProps extends ContainerClassName {
+  participant: Video.Participant;
+  defaultMute: boolean;
+}
+
+export const CameraMicrophoneControls: React.FC<CameraMicrophoneControlsProps> = ({
+  participant,
+  defaultMute,
+  containerClassName,
+}) => {
+  const getChangeStateHandler = useCallback(
+    (variant: "microphone" | "camera"): ((enabled: boolean) => void) => {
+      const tracks =
+        variant === "microphone"
+          ? participant.audioTracks
+          : participant.videoTracks;
+
+      return (enable: boolean) => {
+        tracks.forEach(
+          (track: AudioTrackPublication | VideoTrackPublication) => {
+            const innerTrack = track.track;
+            if (innerTrack) {
+              if (enable && "enable" in innerTrack) innerTrack.enable();
+              else if ("disable" in innerTrack) innerTrack.disable();
+            }
+          }
+        );
+      };
```

Comment:
Since no index is required, one less function with:

```ts
for( const track of tracks) {

}
```

Reply:
That's a matter of choice here and I like `forEach` as a function here. The performance impact is negligible.

Reply:
I've double checked it. You're right on the `forEach` being better choice here https://github.com/sparkletown/sparkle/pull/2231#discussion_r701616807

Just to note, my comment isn't only about performance, generally `forEach` only has one advantage, the `index` argument it provides. So in most cases I recommend `for-of`.